### PR TITLE
Support alternative strace formatting modes

### PR DESCRIPTION
**Pull request by stevenengler**
Changes the option `use_strace_logging: <bool>` to `strace_logging_mode: <"off", "standard", "deterministic">`. The "deterministic" mode hides memory addresses and plugin memory.

### CODE REVIEW

File: src/main/host/syscall/format.rs [hunk 1]
```diff
@@ -77,12 +119,15 @@ macro_rules! simple_pointer_impl {
...
                 &self,
                 f: &mut std::fmt::Formatter<'_>,
+                options: FmtOptions,
                 mem: &MemoryManager,
             ) -> std::fmt::Result {
-                match mem.memory_ref(TypedPluginPtr::new::<$type>(self.ptr, 1)) {
-                    Ok(vals) => write!(f, "{} ({:p})", &(*vals)[0], self.ptr),
+                match (options, mem.memory_ref(TypedPluginPtr::new::<$type>(self.ptr, 1))) {
+                    (FmtOptions::Standard, Ok(vals)) => write!(f, "{} ({:p})", &(*vals)[0], self.ptr),
+                    (FmtOptions::Deterministic, Ok(_vals)) => write!(f, "<pointer>"),
```
Comment (robgjansen) [inline src/main/host/syscall/format.rs:127]: Do we still want the unused `_vals` here?

File: src/main/host/syscall/format.rs [hunk 2]
```diff
@@ -120,12 +169,15 @@ macro_rules! simple_array_impl {
...
                 &self,
                 f: &mut std::fmt::Formatter<'_>,
+                options: FmtOptions,
                 mem: &MemoryManager,
             ) -> std::fmt::Result {
-                match mem.memory_ref(TypedPluginPtr::new::<$type>(self.ptr, K)) {
-                    Ok(vals) => write!(f, "{:?} ({:p})", &(*vals), self.ptr),
+                match (options, mem.memory_ref(TypedPluginPtr::new::<$type>(self.ptr, K))) {
+                    (FmtOptions::Standard, Ok(vals)) => write!(f, "{:?} ({:p})", &(*vals), self.ptr),
+                    (FmtOptions::Deterministic, Ok(_vals)) => write!(f, "<pointer>"),
```
Comment (robgjansen) [inline src/main/host/syscall/format.rs:177]: Do we still want the unused `_vals` here?

**Review by robgjansen — APPROVED** [review]

File: src/main/host/syscall/format.rs (hunk 1 above)
Comment (stevenengler) [reply src/main/host/syscall/format.rs]: Changed to just `_`.

File: src/main/host/syscall/format.rs (hunk 2 above)
Comment (stevenengler) [reply src/main/host/syscall/format.rs]: Changed to just `_`.

**Pull request merged**